src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,60 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from "react";
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  searchInputValue: string;
+  setSearchInputValue: (todos: string) => void;
+  selectedOption: string;
+  setSelectedOption: (option: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  searchInputValue,
+  setSearchInputValue,
+  selectedOption,
+  setSelectedOption,
+}) => {
+
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={selectedOption}
+            onChange={e => setSelectedOption(e.target.value)}
```

Comment:
The naming convention for event handler functions should follow the pattern 'handle' + EventName, e.g., 'handleSelectChange'. Consider renaming 'setSelectedOption' to follow this convention.